solr/solrj/src/java/org/apache/solr/common/cloud/ClusterState.java

```diff
@@ -382,6 +383,7 @@ void setLiveNodes(Set<String> liveNodes) {
    * Be aware that this may return collections which may not exist now. You can confirm that this
    * collection exists after verifying CollectionRef.get() != null
    */
+  @Deprecated // see collectionStream()
```

Comment:
Use `@deprecated` tag in Javadoc instead of dangling comment

Reply:
And IMO this can't be deprecated for performance reasons until `getCollectionNames()` is added to replace it.

Reply:
Indeed; this PR shall not be merged until getCollectionNames is (today!)